dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -264,15 +265,15 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
             scopeExitToCleanRegistrations.release();
 
-            auto lock = s_activatorInfoLock.lock_exclusive();
+            auto lock{ s_activatorInfoLock.lock_exclusive() };
             s_pushTriggerRegistration = registeredTaskFromBuilder;
 
             PushNotificationTelemetry::ActivatorRegisteredByApi(S_OK, details.Activators());
         }
 
         catch(...)
         {
-            HRESULT hrError = wil::ResultFromCaughtException();
```

Comment:
This block (and others like it) should not throw a _new_ exception. Instead, use "re-throw" (just the `throw` keyword.) Like so:

```c++
catch(...)
{
    PushNotificationTelemetry::ActivatorRegisteredByApi(
        wil::ResultFromCaughtException(),
        details ? details.Activators() : PushNotificationRegistrationActivators::Undefined);
    throw;
}
```

That preserves the original exception state while allowing you a chance to clean up.